### ISSUE TITLE
Add Notes and/or Links to Leafs / Graphs

### ISSUE THREAD

**Issue by choefing** [Cacti/cacti#3840]
It would help our operators to have a link to internal procedures or a posibility to add (them in) notes to a graph (or leaf, e.g. on top of all the graphs) like the old [https://docs.cacti.net/userplugin:notices](url) did.

Maybe this can be already achieved by extending "external links" if it would be possible to assign it to a graph id or leaf.
A line breaking dedicated tree item with free text could also be a solution.

If there is already a way to achieve this I was not able to figure out and hints are welcome as well.

Thanks for all the effort.

**Comment by TheWitness** [comment]
Well, it's a plugin that was not merged into core.

**Comment by choefing** [comment]
This is correct as far as I can see but would you consider it helpful and worth beeing merged / added to future releases ?

**Comment by TheWitness** [comment]
Yea, but as you can see, we have a devastating backlog.

**Comment by netniV** [comment]
Is it something that can be more simply updated once off as a plugin for the new stuff?

**Comment by xmacan** [comment]
Little plugins are my forte :-)
If you want plugin, I will take it. For me it is easier to create my own plugin code instead of studying cacti code.

**Comment by TheWitness** [comment]
I was thinking the same thing Petr I just can't get the willpower at the moment to write it.

**Comment by netniV** [comment]
> Little plugins are my forte :-)
> If you want plugin, I will take it. For me it is easier to create my own plugin code instead of studying cacti code.

And then we come along and ask you to change it to our styles ;-)

**Comment by TheWitness** [comment]
Good styles 😁

**Comment by xmacan** [comment]
hehe :-) I was busy, starting to work on in now.

**Comment by xmacan** [comment]
I have few questions:
1) plugin or native cacti code? It is few lines of code ane one small sql table. I prefer cacti code (good styles :-)
2) Now I have only public notice (everyone can change it)
3) display in graph or below? I prefer below.

Screenshots (without css). New icon - Flag
![Screenshot_2020-11-02 Tree Mode xxxx (xxxx - BSD)](https://user-images.githubusercontent.com/26485719/97916895-ee27aa00-1d53-11eb-98f3-2afb733fbc50.png)

Jquery form
![Screenshot_2020-11-02 Tree Mode xxxx (xxxx - BSD)(1)](https://user-images.githubusercontent.com/26485719/97917028-24652980-1d54-11eb-9df3-8f7c9fecce45.png)

**Comment by choefing** [comment]
If that is a wishlist :) well.. lets go.

Depending on how difficult and big it should go, I'd go for including rights management builtin in Cacti.
I dont think anyone needs his/her very "own" note but should be able to add/change/delete it per user/group rights e.g. for the graph to show. If not requested otherwise, I'd go for an implementation of a single note for now to keep it simple.

I don't get how you plan to make it work. Note below graph is like a headline ? Once you click it it shows up the contained information ?
For me it would be really ok to even not show the note directly somewhere but make it accessible with the flag icon ? Having a CSS div popping up or the like ? Even seperate window to keep it in style with the other icons, they also open new site to show e.g. rrd data.

Will those notes beeing added in the graph settings itself ? How woould it work with graph templates then ?
Or is the plan to have like a very own editable window once you click it ?

Personally (if not in use already), I'd go for an "( i )" nformation symbol.

Plugin or included into Cacti ? I really can't tell, lots of plugins have been merged into core already, this could be a point to skip if going directly with core code.

I'm really happy someone thinks my feature request is worth spending time, tea/coffee to work on, really appreciated. Thanks xmacan.


CH

**Comment by netniV** [comment]
Most requests are worth working on, the juggle is deciding which should take prority given complexity, time available and if it works well for the majority.
